packages/playground/data-liberation/src/import/WP_Stream_Importer.php

```diff
@@ -288,14 +296,22 @@ private function next_frontloading_step() {
 		$cursor                            = $this->entity_iterator->get_reentrancy_cursor();
 		$this->active_downloads[ $cursor ] = array();
 
-		$data = $entity->get_data();
+		$data     = $entity->get_data();
+		$upstream = $this->entity_iterator->get_upstream();
+
 		switch ( $entity->get_type() ) {
+			case 'category':
+			case 'term':
+				$this->topological_sorter->map_term( $upstream, $data );
```

Comment:
Note $upstream is very likely long past the opening entity tag. I'm exploring keeping track of that byte offset in https://github.com/WordPress/wordpress-playground/pull/2013 – it should be easy to adjust once that PR lands.

Reply:
Yes, I saw it, thanks. The offset and the length are enough here.